src/Workspaces/VisualBasic/Portable/Indentation/VisualBasicIndentationService.Indenter.vb

```diff
@@ -26,35 +26,40 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Indentation
 
         Protected Overrides Function GetDesiredIndentationWorker(
                 indenter As Indenter,
-                token As SyntaxToken,
-                previousLine As TextLine,
-                lastNonWhitespacePosition As Integer) As IndentationResult
+                tokenOpt As SyntaxToken?,
+                triviaOpt As SyntaxTrivia?) As IndentationResult
 
-            If token.Span.End = lastNonWhitespacePosition + 1 Then
-                Return GetIndentationBasedOnToken(indenter, token)
-            Else
-                Debug.Assert(token.FullSpan.Contains(lastNonWhitespacePosition))
+            If triviaOpt.HasValue Then
+                Dim trivia = triviaOpt.Value
 
-                Dim trivia = indenter.Root.FindTrivia(lastNonWhitespacePosition)
+                If trivia.Kind = SyntaxKind.CommentTrivia OrElse
+                   trivia.Kind = SyntaxKind.DocumentationCommentTrivia Then
 
-                ' preserve the indentation of the comment trivia before a case statement
-                If trivia.Kind = SyntaxKind.CommentTrivia AndAlso trivia.Token.IsKind(SyntaxKind.CaseKeyword) AndAlso trivia.Token.Parent.IsKind(SyntaxKind.CaseStatement) Then
-                    Return indenter.GetIndentationOfLine(previousLine)
+                    ' if the comment is the only thing on a line, then preserve its indentation for the next line.
+                    Dim line = indenter.Text.Lines.GetLineFromPosition(trivia.FullSpan.Start)
+                    If line.GetFirstNonWhitespacePosition() = trivia.FullSpan.Start Then
+                        Return New IndentationResult(trivia.FullSpan.Start, 0)
+                    End If
                 End If
 
-                If trivia.Kind = SyntaxKind.LineContinuationTrivia Then
-                    Return GetIndentationBasedOnToken(indenter, GetTokenOnLeft(trivia), trivia)
-                End If
+                ' preserve the indentation of the comment trivia before a case statement
+                If trivia.Kind = SyntaxKind.CommentTrivia Then
+                    If trivia.Token.IsKind(SyntaxKind.CaseKeyword) AndAlso
+                       trivia.Token.Parent.IsKind(SyntaxKind.CaseStatement) Then
+                        Return New IndentationResult(trivia.SpanStart, 0)
+                    End If
```

Comment:
this part is preserved.  VB indents comments *above* a case statement to that case statement (even though you could argue they belong to the block above it).

Reply:
The scenario here is you already have a first line of a comment, and then adding a second line between the existing comment and the case?

(This seems like a good comment to add into the code rather than the PR here.)

Reply:
will do.

Reply:
Actually, i think this code isn't even necessary.  It should be subsumed now by the above check to align with a standalone comment on the previous line.  testing out.

Reply:
yup.  not necessary. i've removed it.  yaay.